test/cmd/admin/registry.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../../..
+source "${OS_ROOT}/hack/lib/init.sh"
+os::log::stacktrace::install
+trap os::test::junit::reconcile_output EXIT
+
+defaultimage="openshift/origin-\${component}:latest"
+USE_IMAGES=${USE_IMAGES:-$defaultimage}
+
+os::test::junit::declare_suite_start "cmd/admin/registry"
+# Test running a registry as a daemonset
+os::cmd::expect_failure_and_text 'oadm registry --daemonset --dry-run' 'does not exist'
+os::cmd::expect_success_and_text "oadm registry --daemonset -o yaml --credentials=${KUBECONFIG}" 'DaemonSet'
+os::cmd::expect_success "oadm registry --daemonset --credentials=${KUBECONFIG} --images='${USE_IMAGES}'"
+os::cmd::expect_success_and_text 'oadm registry --daemonset' 'service exists'
+os::cmd::expect_success_and_text 'oc get ds/docker-registry --template="{{.status.desiredNumberScheduled}}"' '1'
```

Comment:
Subduers are fine

On Wed, Jul 27, 2016 at 4:49 PM, Steve Kuznetsov <notifications@github.com>
wrote:

> In test/cmd/admin/registry.sh
> <https://github.com/openshift/origin/pull/10065#discussion_r72520628>:
>
> > +
> > +OS_ROOT=$(dirname "${BASH_SOURCE}")/../../..
> > +source "${OS_ROOT}/hack/lib/init.sh"
> > +os::log::stacktrace::install
> > +trap os::test::junit::reconcile_output EXIT
> > +
> > +defaultimage="openshift/origin-\${component}:latest"
> > +USE_IMAGES=${USE_IMAGES:-$defaultimage}
> > +
> > +os::test::junit::declare_suite_start "cmd/admin/registry"
> > +# Test running a registry as a daemonset
> > +os::cmd::expect_failure_and_text 'oadm registry --daemonset --dry-run' 'does not exist'
> > +os::cmd::expect_success_and_text "oadm registry --daemonset -o yaml --credentials=${KUBECONFIG}" 'DaemonSet'
> > +os::cmd::expect_success "oadm registry --daemonset --credentials=${KUBECONFIG} --images='${USE_IMAGES}'"
> > +os::cmd::expect_success_and_text 'oadm registry --daemonset' 'service exists'
> > +os::cmd::expect_success_and_text 'oc get ds/docker-registry --template="{{.status.desiredNumberScheduled}}"' '1'
>
> Nice one
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10065/files/3fc05626580aa600720bac6798d841414b8403cc#r72520628>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_py3gNxLKQG0R8MjRlsDsxUAeu4Fwks5qZ8RagaJpZM4JWmK6>
> .
>